pkg/services/ngalert/metrics/state.go

```diff
@@ -17,5 +19,24 @@ func NewStateMetrics(r prometheus.Registerer) *State {
 			Name:      "alerts",
 			Help:      "How many alerts by state.",
 		}, []string{"state"}),
+		StateUpdateDuration: promauto.With(r).NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: Namespace,
+				Subsystem: Subsystem,
+				Name:      "state_calculation_duration_milliseconds",
+				Help:      "The duration of calculation of a single state",
+				Buckets:   []float64{10, 25, 50, 100, 500, 1000, 10000, 100000},
+			},
+			[]string{"needsImage"},
```

Comment:
I would leave this out for now, `needsImage` can be true even when images are disabled.

Reply:
Right, but if images are enabled the metric will be skewed toward a higher number. It will let us distinguish slowness because of image processing from slow state processing. I wonder if there are better ways